spec/unit/defines/lsstsw_hiera_spec.rb

```diff
@@ -0,0 +1,71 @@
+require 'spec_helper'
+
+# hiera values get "cached" and resued even after hiera_config is changed /
+# unset and will break other spec tests.
+#
+# https://github.com/rodjek/rspec-puppet/issues/215
+module RSpec::Puppet
+  module Support
+    def build_catalog(*args)
+      @@cache[args] = self.build_catalog_without_cache(*args)
+    end
+  end
+end
+
+describe 'lsststack::lsstsw', :type => :define do
+  before(:all) do
+    RSpec.configure do |c|
+      c.hiera_config = 'spec/fixtures/hiera/hiera.yaml'
+    end
+  end
+
+  after(:all) do
+    RSpec.configure do |c|
+      c.hiera_config = '/dev/null'
+    end
+  end
+
+  let(:facts) {{ :osfamily => 'RedHat', :operatingsystemmajrelease => 6 }}
+  let(:name) { 'foo' }
+  let(:title) { name }
+  let(:pre_condition) { 'include ::lsststack' }
+
+  context 'hiera' do
+    context 'data overrides parameters defaults' do
+      hiera_params = {
+        :lsstsw_repo => 'a',
+        :lsstsw_branch => 'b',
+        :lsstsw_ensure => 'latest',
+        :buildbot_repo => 'c',
+        :buildbot_branch => 'd',
+        :buildbot_ensure => 'latest',
+        :lsst_build_repo => 'e',
+        :lsst_build_branch => 'f',
+        :lsst_build_ensure => 'latest',
```

Comment:
I was just wondering about Don't Repeat Yourself. If it's important that what is in the yaml file is meant to match what is in that code then it would be good if the code was explicit about it.